consensus/src/pipeline/pipeline_builder.rs

```diff
@@ -564,11 +565,30 @@ impl PipelineBuilder {
             user_txns.as_ref().clone(),
         ]
         .concat();
+        let proposer_index = block
+            .author()
+            .and_then(|proposer| validator.iter().position(|&v| v == proposer));
+        let auxiliary_info = txns
+            .iter()
+            .map(|txn| {
+                txn.borrow_into_inner().try_as_signed_user_txn().map_or(
```

Comment:
this include metadata txn and validator txn? do we want this for all or only user txn?

Reply:
For priority fee this doesn't matter.